### DISCUSSION TITLE
Ruby install failing on Mac OS 12.3/12.4

### DISCUSSION THREAD

**Discussion by moveson** [rbenv/ruby-build#2006] [Build failures] [answered]
I'm getting an error installing any version of Ruby on an iMac using rbenv:

```
test/v3ext.c:201:24: error: implicitly declaring library function 'memcmp' with type 'int (const void *, const void *, unsigned long)' [-Werror,-Wimplicit-function-declaration]
        if (!TEST_true(memcmp(ip1->data, ip2->data, ip1->length) <= 0))
                       ^
test/v3ext.c:201:24: note: include the header <string.h> or explicitly provide a declaration for 'memcmp'
```
Here's a gist with the full error log:
https://gist.github.com/moveson/55136bc00700a905ece47b368cf00ff1

I've updated the Mac OS and the CLI developer tools, and I've run a full `brew upgrade`. I get the same error regardless of the Ruby version I attempt to install. I've tried 2.7.6, 3.0.4, and 3.1.2.

I've also tried running the install with `RUBY_CFLAGS="-Wno-error=implicit-function-declaration"` as recommended [here](https://stackoverflow.com/questions/70229447/inspect-or-clean-up-the-working-tree-when-installing-ruby-on-mac-m1)

**Comment by andre-alves** [comment, marked as answer]
I think you are using the incorrect environment. Try:

`OPENSSL_CFLAGS=-Wno-error=implicit-function-declaration`

[Source](https://github.com/rbenv/ruby-build/issues/1998#issuecomment-1180698666).

**Reply by moveson** [reply]
That worked. Thank you!
